includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding\Services;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Options;
+use NewfoldLabs\WP\Module\Onboarding\Data\Flows;
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+class FlowService {
+
+	public static function initalize_flow_data() { 
+         if ( ! ( $flow_data = self::read_flow_data_from_wp_option() ) ) {
+            $flow_data =  self::get_default_flow_data();
+			return \update_option( Options::get_option_name( 'flow' ), $flow_data );;
+		}
+		$default_flow_data = self::get_default_flow_data();
+		$updated_flow_data = self::update_flow_data_recursive($default_flow_data, $flow_data);
+		return \update_option( Options::get_option_name( 'flow' ), $updated_flow_data );
+    }
+
+    public static function get_default_flow_data() {
+		// check if data is available in the database if not then fetch the default dataset
+		$flow_data              = Flows::get_data();
+		$flow_data['createdAt'] = time();
+		// get current flow type
+		$flow_type = Data::current_flow();
+		// update default data if flow type is ecommerce
+		if($flow_type == 'ecommerce') 
+			$flow_data = self::update_default_data_for_ecommerce( $flow_data );
+		return $flow_data;
+	}
+
+	public static function update_flow_data($params) {
+		$flow_data = FlowService::read_flow_data_from_wp_option();
+		return self::update_post_call_data_recursive($flow_data, $params);
+	}
+
+	/*
+	 * function to update the Flow Data (Blueprint) in an array recursively in comparison to Flows::get_data() (Database)
+	 */	
+	private static function update_flow_data_recursive($default_flow_data, $flow_data) {
+		$flow_data_fixes = Flows::get_fixes();
+		$exception_list = Flows::get_exception_list();
+		$updated_flow_data = array();
+		foreach ($default_flow_data as $key => $value)
+		{ 
+			// To align the datatype of the respective values with the one set in the blueprint
+			if(strcmp(gettype($value), gettype($flow_data[$key])) != 0) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Any Key renamed is updated in the database with NewKey and the value from the OldKey is retained or not based on retain_existing_value
+			if($flow_data_fixes) {
+				foreach ($flow_data_fixes as $old_key=>$replace_key) {
+					if (array_key_exists($old_key, $flow_data) && array_key_exists($replace_key['new_key'], $default_flow_data)) {
+						($replace_key['retain_existing_value'])?
+							$flow_data[$replace_key['new_key']] = $flow_data[$old_key]
+							: $flow_data[$replace_key['new_key']] = $default_flow_data[$replace_key['new_key']];
+						unset($flow_data[$old_key]);
+					}	
```

Comment:
Here, the correct value is being replaced in `$flow_data`, which is the database of the user. We are populating the updated data in `$updated_flow_data`, which is comparing and taking the values from `$default_flow_data` and `$flow_data` . Hence, `$flow_data` needs to be updated first with the right key-value from the fixes and then followed by populating the right key-values to `$updated-flow_data`

Reply:
Exactly my point can't we just make the update to `$updated_flow_data` instead of `$flow_data` and skip the rest since this is also comparing with `$default_flow_data` on line 58?